mm2src/coins/hd_wallet/mod.rs

```diff
@@ -241,20 +242,61 @@ where
     }
 }
 
-pub async fn load_hd_accounts_from_storage<HDAddress, ExtendedPublicKey>(
+pub async fn load_hd_accounts_from_storage_with_matching_xpubs<HDAddress>(
     hd_wallet_storage: &HDWalletCoinStorage,
     derivation_path: &HDPathToCoin,
-) -> HDWalletStorageResult<HDAccountsMap<HDAccount<HDAddress, ExtendedPublicKey>>>
+    global_hd_ctx: &GlobalHDAccountArc,
```

Comment:
I think we should separate the HD wallet implementation from `GlobalHDAccountCtx`, so we shouldn't pass this here.